### ISSUE TITLE
InvalidOperationException in Examine.LuceneEngine.OpenReaderTracker.CloseStaleReaders

### ISSUE THREAD

**Issue by max-maurice** [Shazwazza/Examine#220]
Calling a second time IndexRebuilder.RebuildIndex seems to throw this exception:

```   
System.InvalidOperationException: Die Warteschlange ist leer.
bei System.ThrowHelper.ThrowInvalidOperationException(ExceptionResource resource)
   bei Examine.LuceneEngine.OpenReaderTracker.CloseStaleReaders(Directory dir, TimeSpan ts) in C:\projects\examine-qvx04\src\Examine\LuceneEngine\OpenReaderTracker.cs:Zeile 56.
   bei Examine.LuceneEngine.Providers.LuceneIndex.EnsureIndex(Boolean forceOverwrite) in C:\projects\examine-qvx04\src\Examine\LuceneEngine\Providers\LuceneIndex.cs:Zeile 437.
   bei Examine.LuceneEngine.Providers.LuceneIndex.CreateIndex() in C:\projects\examine-qvx04\src\Examine\LuceneEngine\Providers\LuceneIndex.cs:Zeile 507.
   bei Umbraco.Examine.IndexRebuilder.RebuildIndex(String indexName) in D:\a\1\s\src\Umbraco.Examine\IndexRebuilder.cs:Zeile 43.

[...]

```

("queue is empty")

This didn't happen with Examine 1.0.2 and it seems it is connected to 06cf3d0e17771412d21aec7a003a50cee5ff3ab0 (#209).

I'm not sure, but maybe it could help to check the queue size before calling Peek() in Examine.LuceneEngine.OpenReaderTracker.CloseStaleReaders().

**Comment by Shazwazza** [comment]
Thanks for reporting :) IndexRebuilder is part of Umbraco not Examine directly but I'll check it out to see if I can replicate in a test in this proj.

**Comment by Shazwazza** [comment]
https://github.com/umbraco/Umbraco-CMS/issues/10273

**Comment by Shazwazza** [comment]
Fixed in rev 28c6b32b390c15f7959ae5a2f22e68b7ad69fd59

**Comment by p-m-j** [comment]
> Fixed in rev [28c6b32](https://github.com/Shazwazza/Examine/commit/28c6b32b390c15f7959ae5a2f22e68b7ad69fd59)

That sha is really confusing as it only shows gemfile.lock changes, guessing should be 6b74baf3533f68bce87b2fa2c65be61fc87c7d2f
